sdk/identity/src/device_code_flow/mod.rs

```diff
@@ -171,7 +171,7 @@ async fn post_form(
     url: &str,
     form_body: String,
 ) -> Result<Response> {
-    let url = Request::parse_uri(url)?;
+    let url = Url::parse(url).map_kind(ErrorKind::DataConversion)?;
```

Comment:
`map_kind` not needed. 

https://github.com/Azure/azure-sdk-for-rust/blob/02e025a49c1e081d2be7ac4e57287dd0727e551b/sdk/core/src/error/mod.rs#L236-L240